tools/gfx/persistent-shader-cache.cpp

```diff
@@ -96,13 +132,17 @@ LinkedNode<ShaderCacheEntry>* PersistentShaderCache::findEntry(const slang::Dige
         entries.AddFirst(entryNode);
         if (mutableShaderCacheFileSystem)
         {
-            saveCacheToFile();
+            auto lastAccessedString = String(duration_cast<duration<double>>(high_resolution_clock::now().time_since_epoch()).count());
```

Comment:
You can store a `List<ShaderCacheEntry> ` that is exactly the same as the whole file in memory.
Your linked list can just be `LinkedList<ShaderCacheEntry*>` or `LinkedList<Index>` that points to locations in the `List`. The offset can then be calculated directly without needing the `entryToOffset` dictionary.

Reply:
done